src/go/configgenerator/listener_generator.go

```diff
@@ -50,8 +50,87 @@ const (
 	statPrefix = "ingress_http"
 )
 
+// MakeListeners provides dynamic listeners for Envoy
+func MakeListeners(serviceInfo *sc.ServiceInfo) ([]*v2pb.Listener, error) {
+	var listeners []*v2pb.Listener
+
+	port := serviceInfo.Options.ListenerPort
+	isSsl := false
+	if serviceInfo.Options.SslPort != 0 {
+		port = serviceInfo.Options.SslPort
+		isSsl = true
+	}
+
+	listener, err := makeListener(serviceInfo, port, isSsl)
```

Comment:
Add a IsSsl  and getSslPort method to serviceInfo instead of assign a temporary variable, so makeListener only need one arg `serviceInfo`

Reply:
Done. Just move this into makeListener.